test/services/BaseRepositoryTest.java

```diff
@@ -516,7 +517,7 @@ public void testRankKeyword() throws IOException, InterruptedException {
   @Test
   public void testSearchBySubjectClassification() throws IOException, InterruptedException {
     Resource db1 = getResourceFromJsonFile("BaseRepositoryTest/testSearchBySubjectClassification.DB.1.json");
-    mBaseRepo.addResource(db1, mMetadata);
+    mBaseRepo.importResources(Arrays.asList(db1), mMetadata);
```

Comment:
@philboeselager the problem was not the framer, but the way the test data was imported. When using `addResource`, only the resource itself (i.e. its [concise bounded description](https://www.w3.org/Submission/CBD/)) is added to to triple store, any embedded resources (such as the classifications) are discarded. Using `importResources` also extracts embedded resources and adds them to the triple store so that they are available when framing.